x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/inference/rerank/RerankOperator.java

```diff
@@ -79,8 +66,7 @@ public String toString() {
      */
     @Override
     protected RerankOperatorRequestIterator requests(Page inputPage) {
-        int inputBlockChannel = inputPage.getBlockCount() - 1;
-        return new RerankOperatorRequestIterator(inputPage.getBlock(inputBlockChannel), inferenceId(), queryText, batchSize);
+        return new RerankOperatorRequestIterator((BytesRefBlock) rowEncoder.eval(inputPage), inferenceId(), queryText, batchSize);
```

Comment:
is it always a `BytesRefBlock` ? probably it is, but I wonder if we should be safe and do a check here?